usd/src/sdf_parser/Geometry.cc

```diff
@@ -0,0 +1,460 @@
+      for (unsigned int j = 0; j < subMesh->TexCoordCount(); ++j)
+      {
+        const auto &uv = subMesh->TexCoord(j);
+        uvs.push_back(pxr::GfVec2f(uv[0], 1 - uv[1]));
+      }
+
+      // copy the submesh's normals
+      for (unsigned int j = 0; j < subMesh->NormalCount(); ++j)
+      {
+        const auto &normal = subMesh->Normal(j);
+        normals.push_back(pxr::GfVec3f(normal[0], normal[1], normal[2]));
+      }
+
+      // set the usd mesh's "faceVertexCounts" array according to
+      // the submesh primitive type
+      // TODO(adlarkin) support all primitive types. The computations are more
+      // involved for LINESTRIPS, TRIFANS, and TRISTRIPS. I will need to spend
+      // some time deriving what the number of faces for these primitive types
+      // are, given the number of indices. The "faceVertexCounts" array will
+      // also not have the same value for every element in the array for these
+      // more complex primitive types (see the TODO note in the for loop below)
+      unsigned int verticesPerFace = 0;
+      unsigned int numFaces = 0;
+      switch (subMesh->SubMeshPrimitiveType())
+      {
+        case ignition::common::SubMesh::PrimitiveType::POINTS:
+          verticesPerFace = 1;
+          numFaces = subMesh->IndexCount();
+          break;
+        case ignition::common::SubMesh::PrimitiveType::LINES:
+          verticesPerFace = 2;
+          numFaces = subMesh->IndexCount() / 2;
+          break;
+        case ignition::common::SubMesh::PrimitiveType::TRIANGLES:
+          verticesPerFace = 3;
+          numFaces = subMesh->IndexCount() / 3;
+          break;
+        case ignition::common::SubMesh::PrimitiveType::LINESTRIPS:
+        case ignition::common::SubMesh::PrimitiveType::TRIFANS:
+        case ignition::common::SubMesh::PrimitiveType::TRISTRIPS:
+        default:
+          std::cerr << "Submesh " << subMesh->Name()
+                    << " has a primitive type that is not supported.\n";
+          return false;
+      }
+      // TODO(adlarkin) update this loop to allow for varying element
+      // values in the array (see TODO note above). Right now, the
+      // array only allows for all elements to have one value, which in
+      // this case is "verticesPerFace"
+      for (unsigned int n = 0; n < numFaces; ++n)
+        faceVertexCounts.push_back(verticesPerFace);
+
+      std::string primName;
+      if (!subMesh->Name().empty())
+        primName = _path + "/" + subMesh->Name();
+      else
+        primName = _path + "/submesh_" + std::to_string(i);
+
+      primName = removeDash(primName);
+
+      auto usdMesh = pxr::UsdGeomMesh::Define(_stage, pxr::SdfPath(primName));
+      usdMesh.CreatePointsAttr().Set(meshPoints);
+      usdMesh.CreateFaceVertexIndicesAttr().Set(faceVertexIndices);
+      usdMesh.CreateFaceVertexCountsAttr().Set(faceVertexCounts);
+
+      auto coordinates = usdMesh.CreatePrimvar(
+          pxr::TfToken("st"), pxr::SdfValueTypeNames->Float2Array,
```

Comment:
Is there a reason why the token here is `st`? This is a PrimVar for UV coordinates, so would it be better to set the token to `uv`?

Reply:
it's just a different naming convention.

uv coordinates start from the upper left corner (v-axis is facing down).
st coordinates start from the lower left corner (t-axis is facing up).
```
s = u;
t = 1-v;
```